In [1]:
import numpy as np
import scipy.stats as stats

# Семинар 7

## Задача 1

Даны две  независимые выборки. Не соблюдается условие нормальности

x1  380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

In [2]:
x1 = np.array([380,420, 290])
y1 = np.array([140,360,200,900])

In [3]:
stats.mannwhitneyu(x1,y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Т.к. pvalue = 0.63 > alpha = 0.05 можно сделать вывод, что статистически значимых различий нет (нулевая гипотеза верна)

## Задача 2

Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

In [4]:
before = np.array([150, 160, 165, 145, 155])
second =np.array([140, 155, 150,  130, 135])
third = np.array([130, 130, 120, 130, 125])

In [5]:
stats.friedmanchisquare(before, second, third)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Т.к. pvalue = 0.008 < alpha = 0.05, можно сделать вывод, что статистически значимые различия есть (альтернативная гипотеза верна)

## Задача 3

Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [6]:
stats.wilcoxon(before,second)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Т.к. pvalue = 0.63 > alpha = 0.05 можно сделать вывод, что статистически значимых различий нет (нулевая гипотеза верна)

## Задача 4

Даны 3 группы  учеников плавания.

В 1 группе время на дистанцию 50 м составляют:

56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [7]:
group_1 =np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
group_2 =np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
group_3 =np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [8]:
stats.kruskal(group_1, group_2, group_3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

Т.к. pvalue = 0.65 > alpha = 0.05 можно сделать вывод, что статистически значимых различий нет (нулевая гипотеза верна)

## Задача 4

Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [9]:
x = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
mu = 2.5
n=10
alpha = 0.05

In [10]:
# Среднее арифметическое выборки
X = np.mean(x)
X

2.5279999999999996

In [11]:
std = np.std(x, ddof=1)
std

0.1572542173961923

In [12]:
t = (X-mu) / (std/n**0.5)
t

0.5630613661802959

In [13]:
# Вероятность для получения табличного t
1-0.05/2

0.975

t табличное для p  =  0.975 и двустороннего теста при степени свободы ( df  =  10 - 1) будет  2.262

Т.к. t табличное  = 2.262 и больше расчетного t = 0.563 можно сделать вывод, что  Нулевая гипотеза  верна